Gems/EMotionFX/Code/Tests/UI/SkeletonOutlinerTestFixture.cpp

```diff
@@ -0,0 +1,172 @@
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#include <Editor/ColliderContainerWidget.h>
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include "SkeletonOutlinerTestFixture.h"
+#include <Editor/ColliderHelpers.h>
+#include <Editor/ObjectEditor.h>
+#include <AzToolsFramework/UI/PropertyEditor/ReflectedPropertyEditor.hxx>
+#include <AzToolsFramework/UI/PropertyEditor/PropertyRowWidget.hxx>
+#include <AzToolsFramework/UI/PropertyEditor/PropertyDoubleSpinCtrl.hxx>
+#include <AzQtComponents/Components/Widgets/SpinBox.h>
+
+#include <Tests/TestAssetCode/TestActorAssets.h>
+#include <Tests/TestAssetCode/SimpleActors.h>
+
+#include <QApplication>
+#include <QTest>
+
+namespace EMotionFX
+{
+    void SkeletonOutlinerTestFixture::SetUpPhysics()
+    {
+        EMStudio::GetMainWindow()->ApplicationModeChanged("Physics");
+        const int numJoints = 6;
+        AZ::Data::AssetId actorAssetId("{5060227D-B6F4-422E-BF82-41AAC5F228A5}");
+        AZ::Data::Asset<Integration::ActorAsset> actorAsset =
+            TestActorAssets::CreateActorAssetAndRegister<SimpleJointChainActor>(actorAssetId, numJoints, "ClothColliderTestsActor");
+
+        CreateSkeletonAndModelIndices();
+
+        EXPECT_EQ(m_indexList.size(), numJoints);
+    }
+    void SkeletonOutlinerTestFixture::AddColliderViaAddComponentButton(QString label, QString subLevelLabel)
+    {
+        EXPECT_GT(m_indexList.size(), 3) << "Make sure to have a skeleton";
+        // Find the 3rd joint after the RootJoint in the TreeView and select it
+        SelectIndexes(m_indexList, m_treeView, 3, 3);
+
+        auto* treeView = GetAddCollidersTreeView();
+        auto* model = treeView->model();
+
+        //  find indices
+        QModelIndexList indices = model->match(
+            model->index(0, 0),
+            Qt::DisplayRole,
+            QVariant::fromValue(label),
+                    -1,
+            Qt::MatchRecursive);
+
+        if (subLevelLabel != "")
+        {
+            indices = model->match(
+                model->index(0, 0, indices[0]),
+                Qt::DisplayRole,
+                QVariant::fromValue(subLevelLabel),
+                    -1,
+                Qt::MatchRecursive);
+        }
+        //  check indices
+        if (indices.size() < 1)
+        {
+            EXPECT_GT(indices.size(), 0) << "Label not found";
+            return;
+        }
+        EXPECT_EQ(indices.size(), 1) << "Label is not eindeutig";
+
+        //  click first index
+        auto index = indices[0];
+        treeView->selectionModel()->select(index, QItemSelectionModel::Select | QItemSelectionModel::Rows);
+        treeView->clicked(index);
+    }
+
+    void SkeletonOutlinerTestFixture::ShowJointPropertyWidget()
+    {
+        auto* widget = GetJointPropertyWidget();
+        auto* mw = new QMainWindow;
```

Comment:
same about the auto. Also I think it's best for readability to use mainWindow instead of mw.